tests/RobotFramework/tests/cumulocity/registration/device_registration.robot

```diff
@@ -161,6 +161,15 @@ Register tedge-log-plugin when tedge-mapper-c8y is not running #2389
     Should Have MQTT Messages    te/device/offlinechild3///cmd/log_upload/+
 
 
+Auto register disabled
+    Execute Command    sudo tedge config set c8y.entity_store.auto_register false
+    Restart Service    tedge-mapper-c8y    
+    Service Health Status Should Be Up    tedge-mapper-c8y    
+    Execute Command    sudo tedge mqtt pub 'te/device/${CHILD_SN}///a/temperature_high' '{ "severity": "critical", "text": "Temperature is very high", "time": "2021-01-01T05:30:45+00:00" }' -q 2 -r
+    ${result}=    Execute Command    tedge mqtt sub te/errors & sleep 4s; kill $!    ignore_exit_code=${True}    stdout=${True}    stderr=${False}
```

Comment:
Spawning the subscriber after sending the measurement is fragile as the error message may have been sent before this subscriber is started. Why not just use `Should Have MQTT Messages` itself?

Reply:
As you can see the test failed in the workflow, presumably for that exact same reason.

Reply:
fixed